src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -317,8 +317,15 @@ private static String debugString(Object arg) {
      * Extract the MemberName of a newly-defined method.
      */
     private MemberName loadMethod(byte[] classFile) {
+        Object cd;
+        if (classData.size() == 1) {
+            cd = classData.get(0).value;
+        } else {
+            cd = classDataValues();
+        }
+
         Class<?> invokerClass = LOOKUP.makeHiddenClassDefiner(className, classFile, Set.of())
-                                      .defineClass(true, classDataValues());
```

Comment:
Why not just put it `classDataValues()` instead?

```
    Object classDataValues() {
        final List<ClassData> cd = classData;
        return switch(cd.size()) {
            case 0 -> null; // flatten for zero case
            case 1 -> cd.get(0).value; // flatten for single value case
            case 2 -> List.of(cd.get(0).value, cd.get(1).value);
            ...
```

And it also covers zero case for free.

Reply:
I fixed it as you suggested. I also changed the method to private to make sure no one else is using this method.